google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableCloudMonitoringExporter.java

```diff
@@ -121,48 +130,64 @@ public CompletableResultCode export(Collection<MetricData> collection) {
     }
     if (!collection.stream()
         .flatMap(metricData -> metricData.getData().getPoints().stream())
-        .allMatch(pd -> projectId.equals(BigtableExporterUtils.getProjectId(pd)))) {
+        .allMatch(pd -> bigtableProjectId.equals(BigtableExporterUtils.getProjectId(pd)))) {
       logger.log(Level.WARNING, "Metric data has different a projectId. Skip exporting.");
       return CompletableResultCode.ofFailure();
     }
 
-    List<TimeSeries> allTimeSeries;
+    List<TimeSeries> bigtableTimeSeries;
     try {
-      allTimeSeries =
-          BigtableExporterUtils.convertCollectionToListOfTimeSeries(
-              collection, taskId, monitoredResource);
+      bigtableTimeSeries =
+          BigtableExporterUtils.convertToBigtableTimeSeries(
+              collection.stream()
+                  .filter(
+                      md ->
+                          !md.getName()
+                              .contains(BuiltinMetricsConstants.PER_CONNECTION_ERROR_COUNT_NAME))
+                  .collect(Collectors.toList()),
+              taskId);
     } catch (Throwable e) {
-      logger.log(Level.WARNING, "Failed to convert metric data to cloud monitoring timeseries.", e);
+      logger.log(
+          Level.WARNING,
+          "Failed to convert bigtable metric data to cloud monitoring timeseries.",
+          e);
       return CompletableResultCode.ofFailure();
     }
 
-    ProjectName projectName = ProjectName.of(projectId);
-    CreateTimeSeriesRequest request =
+    ProjectName projectName = ProjectName.of(bigtableProjectId);
+    CreateTimeSeriesRequest bigtableRequest =
         CreateTimeSeriesRequest.newBuilder()
             .setName(projectName.toString())
-            .addAllTimeSeries(allTimeSeries)
+            .addAllTimeSeries(bigtableTimeSeries)
             .build();
 
-    ApiFuture<Empty> future = this.client.createServiceTimeSeriesCallable().futureCall(request);
-
-    lastExportCode = new CompletableResultCode();
+    ApiFuture<Empty> future =
+        this.client.createServiceTimeSeriesCallable().futureCall(bigtableRequest);
 
+    CompletableResultCode bigtableExportCode = new CompletableResultCode();
     ApiFutures.addCallback(
         future,
         new ApiFutureCallback<Empty>() {
           @Override
           public void onFailure(Throwable throwable) {
-            logger.log(Level.WARNING, "createServiceTimeSeries request failed. ", throwable);
-            lastExportCode.fail();
+            logger.log(
+                Level.WARNING,
+                "createServiceTimeSeries request failed for bigtable metrics. ",
+                throwable);
+            bigtableExportCode.fail();
           }
 
           @Override
           public void onSuccess(Empty empty) {
-            lastExportCode.succeed();
+            bigtableExportCode.succeed();
           }
         },
         MoreExecutors.directExecutor());
 
+    CompletableResultCode gceOrGkeExportCode = exportGceOrGkeMetrics(collection);
```

Comment:
Can you extract the resource level metrics to a function as well so that this function can focus on composing the futures?